### PR TITLE
Feature/error component

### LINKED ISSUE #634: Error component
## Overview
Create a error component to prevent copying the same content and style at many places.

## Acceptance criteria
- [x] Create an error component (may take some options/input to allow different behavior)
- [x] use it everywhere it is applicable
- [x] test it is carefully, force errors on every changed component to check that it works as expected

### PR DESCRIPTION
**Pull request by Iloveall**
Fixes #634

### CODE REVIEW

**Comment by ThomasJuster** [comment]
## Tests

- [ ] src/app/core/components/left-nav/left-nav.component.html
- [ ] src/app/core/components/left-nav/left-nav.component.scss
- [ ] src/app/modules/group/components/associated-activity/associated-activity.component.html
- [ ] src/app/modules/group/components/group-log-view/group-log-view.component.html
- [ ] src/app/modules/group/components/group-manager-list/group-manager-list.component.html
- [ ] src/app/modules/group/components/group-remove-button/group-remove-button.component.html
- [ ] src/app/modules/group/components/group-remove-button/group-remove-button.component.scss
- [ ] src/app/modules/group/components/joined-group-list/joined-group-list.component.html
- [ ] src/app/modules/group/components/managed-group-list/managed-group-list.component.html
- [ ] src/app/modules/group/components/member-list/member-list.component.html
- [ ] src/app/modules/group/components/member-list/member-list.component.scss
- [ ] src/app/modules/group/components/suggestion-of-activities/suggestion-of-activities.component.html
- [ ] src/app/modules/group/pages/group-access/group-access.component.html
- [ ] src/app/modules/group/pages/group-details/group-details.component.html
- [ ] src/app/modules/group/pages/group-edit/group-edit.component.html
- [ ] src/app/modules/group/pages/user/user.component.html
- [ ] src/app/modules/item/components/chapter-children/chapter-children.component.html
- [ ] src/app/modules/item/components/parent-skills/parent-skills.component.html
- [ ] src/app/modules/item/components/sub-skills/sub-skills.component.html
- [ ] src/app/modules/item/pages/chapter-group-progress/chapter-group-progress.component.html
- [ ] src/app/modules/item/pages/chapter-user-progress/chapter-user-progress.component.html
- [ ] src/app/modules/item/pages/group-progress-grid/group-progress-grid.component.html
- [ ] src/app/modules/item/pages/item-log-view/item-log-view.component.html

File: src/app/modules/shared-components/components/error/error.component.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div class="wrapper">
```
Comment (ThomasJuster) [inline src/app/modules/shared-components/components/error/error.component.html:1]: I found other components that should probably use the `<alg-error />` component, can you check if it is indeed the case or not :pray: ?

- [ ] item-children-edit.component.html #l113 ([test link](https://dev.algorea.org/branch/feature/error-component/en/#/activities/by-id/1980584647557587953;path=4702,4102;attempId=0/edit))
- [ ] item-remove-button.component.html #l17 ([test link](https://dev.algorea.org/branch/feature/error-component/en/#/activities/by-id/1980584647557587953;path=4702,4102;attempId=0/edit/advanced-parameters))
- [ ] item-by-id.component.html #l10 ([test link](https://dev.algorea.org/branch/feature/error-component/en/#/activities/by-id/39530140456452546;path=4702,4102,1980584647557587953;attempId=0/edit/advanced-parameters))
- [ ] pending-requests.component.html #l12 ([test link](https://dev.algorea.org/branch/feature/error-component/en/#/groups/mine), expand "pending request" or "pending invitations")

**Review by ThomasJuster — CHANGES_REQUESTED** [review]

File: src/app/modules/item/pages/item-log-view/item-log-view.component.html [hunk 2]
```diff
@@ -1,11 +1,13 @@
 <ng-container *ngIf="state$ | async as state">
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/item/pages/item-log-view/item-log-view.component.html:1]: Fixed

File: src/app/modules/item/pages/group-progress-grid/group-progress-grid.component.html [hunk 3]
```diff
@@ -4,11 +4,13 @@
 ></alg-composition-filter>
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/item/pages/group-progress-grid/group-progress-grid.component.html:4]: Fixed

File: src/app/modules/shared-components/components/error/error.component.html (hunk 1 above)
Comment (Iloveall) [reply src/app/modules/shared-components/components/error/error.component.html:1]: item-children-edit.component.html - Fixed
pending-requests.component.html - Fixed
item-remove-button.component.html - Hmm, I've add <alg-error there
item-by-id.component.html - I guess there is too specific template

**Review by ThomasJuster — APPROVED** [review]

**Pull request merged**